### PR TITLE
Disallow fallback kernel in FakeTensor by default

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #97715

### CODE REVIEW

File: torch/_subclasses/fake_tensor.py
Comment (ezyang) [inline torch/_subclasses/fake_tensor.py:1240]: you mean prims, not prim. Also, we shouldn't allow fallback on prims anyway, they should ALWAYS have meta by construction

**Review by ezyang — APPROVED** [review]
looks fine but please fix the glaring bug

**Review by eellison — COMMENTED** [review]
What is the advantage of doing this instead of changing the default of `allow_fallback_kernels` to False ? Now there is no way for a user to override this behavior.

**Comment by ezyang** [comment]
> What is the advantage of doing this instead of changing the default of allow_fallback_kernels to False ? Now there is no way for a user to override this behavior.

It's a question of defaults. By doing it this way, we guarantee that we don't IMA on indexy custom ops.

**Comment by eellison** [comment]
Yea, understood - but why don't we just switch the default to False then ? Instead of this proposed change

**Comment by ezyang** [comment]
I think I proposed this to you and you didn't like it because it would have reduced fake tensor coverage. For the record, I ALSO support changing this default.

**Comment by eellison** [comment]
Ugh I think I prefer that change to this one. but i'll let you make the final call.

**Review by zou3519 — APPROVED** [review]
LGTM modulo Ed's comment

**Comment by tugsbayasgalan** [comment]
Going to set allow_fallback_kernel to False by default.

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
But I guess we may have to add all of the missing metas now up front :(

**Pull request closed without merging**
